presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedQueryResource.java

```diff
@@ -156,4 +158,23 @@ private void assertStateCounts(List<BasicQueryInfo> infos, int expectedFinished,
         assertEquals(running, expectedRunning);
         assertEquals(queued, expectedQueued);
     }
+
+    @Test
+    public void testGetAllQueryInfoForLimits() throws InterruptedException
+    {
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf100.orders");
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf100.orders");
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf101.orders");
+        runToQueued(client, coordinator1, "SELECT * from tpch.sf102.orders");
+
+        List<BasicQueryInfo> queries;
+        do {
+            MILLISECONDS.sleep(100);
+            queries = resourceManager.getClusterStateProvider().getClusterQueries();
+        } while (queries.size() != 4);
+
+        List<BasicQueryInfo> infos = getQueryInfos(client, coordinator1, "/v1/query?limit=2");
```

Comment:
instead of submitting 4 queries do it for 5. First 3 for running and the last 2 in queued state. And then fetch the result with limit = 4. This should return first 3 queries being running and the last one in queue state. And that will cover all use cases that we are looking for.